src/main.rs

```diff
@@ -758,6 +780,58 @@ async fn get_avl_head(State(state): State<Arc<AppState>>) -> impl IntoResponse {
     }
 }
 
+#[inline(always)]
+async fn get_head(
+    Path(chain_id): Path<u64>,
+    State(state): State<Arc<AppState>>,
+) -> impl IntoResponse {
+    match state.chains.get(&chain_id) {
```

Comment:
Have you considered using the same approach as a task `track_slot_avail_task`?

Reply:
I'd rather not as we need latest state i.e. no delay.

Reply:
We can reduce the cache time. What does the second part mean, I didn't understand.

Reply:
My point was to avoid fetching a proof for a chain that isn't yet in sync. To achieve this, the request for the head from the contract must first be initiated. A task like track_slot_avail_task would simply read the state from an internal cache.